service/acm/api.go

```diff
@@ -107,6 +108,25 @@ func (c *ACM) AddTagsToCertificate(input *AddTagsToCertificateInput) (*AddTagsTo
 	return out, err
 }
 
+// AddTagsToCertificateEx is the same as AddTagsToCertificate with the addition of
+// the ability to pass a context and additional request options.
+//
+// See AddTagsToCertificate for details on how to use this API operation.
+//
+// The context will be used to cancel the request, and any retry delays. The
+// underlying request's retries may be given sub contexts in the future.
+// See the aws/request package's Request and Option types for configurations.
+//
+// If the context is nil, it will be ignored by the API operation.
+func (c *ACM) AddTagsToCertificateWithContext(ctx aws.Context, input *AddTagsToCertificateInput, opts ...request.Option) (*AddTagsToCertificateOutput, error) {
+	req, out := c.AddTagsToCertificateRequest(input)
+	if ctx != nil {
```

Comment:
Checks like this should not be required. When a context is provided, it should be non-nil (per https://golang.org/pkg/context/).